src/client/components/app/Transactions/RepayPositionTx.tsx

```diff
@@ -153,25 +185,37 @@ export const RepayPositionTx: FC<RepayPositionProps> = (props) => {
 
   useEffect(() => {
     // populate collateral reservesmap in redux state
-    if (Object.keys(reservesMap).length === 0) {
-      // get all collateral tokens of type revenue
-      const lineRevenueSummary = selectedLine!.spigot!.revenueSummary;
-      const revenueCollateralTokens: RevenueSummary[] = Object.values(lineRevenueSummary).filter(
-        (token) => token.type === 'revenue'
+    // get all collateral tokens of type revenue
+    const lineRevenueSummary = selectedSpigot!.revenueSummary;
+    const revenueCollateralTokens: RevenueSummary[] = Object.values(lineRevenueSummary).filter(
+      (token) => token.type === 'revenue'
+    );
+
+    // get all credit tokens
+    const creditTokens = Object.values(positions).map((position) => position.token);
+
+    // populate reserves map with each revenue collateral token
+    for (const token of revenueCollateralTokens as RevenueSummary[]) {
+      dispatch(
+        CollateralActions.tradeable({
+          lineAddress: getAddress(selectedPosition!.line),
+          spigotAddress: getAddress(selectedLine!.spigotId),
+          tokenAddress: getAddress(token.token.address),
+          network: walletNetwork!,
+        })
       );
+    }
 
-      // populate reserves map with each revenue collateral token
-      for (const token of revenueCollateralTokens as RevenueSummary[]) {
-        console.log('revenue token: ', token);
-        dispatch(
-          CollateralActions.tradeable({
-            lineAddress: getAddress(selectedPosition!.line),
-            spigotAddress: getAddress(selectedLine!.spigotId),
-            tokenAddress: getAddress(token.token.address),
-            network: walletNetwork!,
-          })
-        );
-      }
+    // populate reserves map with each credit token
+    for (const token of creditTokens) {
```

Comment:
should merge the creditToken and revenueToken lists since they might overlap and then get tradable

Reply:
fixed to merge lists and only keep unique values